common/zoom/zoom-scheduled-meeting.ts

```diff
@@ -204,4 +204,47 @@ const getZoomMeetingReport = async (meetingId: string) => {
     return response.json();
 };
 
-export { getZoomMeeting, getUsersZoomMeetings, createZoomMeeting, deleteZoomMeeting, getZoomMeetingReport };
+const updateZoomMeeting = async (meetingId: string, startTime?: Date, duration?: number, endTime?: Date): Promise<void> => {
+    assureZoomFeatureActive();
+
+    const { access_token } = await getAccessToken();
+    const tz = 'Europe/Berlin';
+    const start = moment(startTime).tz(tz).format('YYYY-MM-DDTHH:mm:ss');
+    const end = moment(endTime).tz(tz).format('YYYY-MM-DDTHH:mm:ss');
+
+    const body = JSON.stringify({
+        start_time: start,
+        duration: duration,
+        timezone: tz,
+        recurrence: {
+            end_date_time: end,
+            type: RecurrenceMeetingTypes.WEEKLY,
+        },
+    });
+
+    logger.info(`UPDATE BODY: ${body}`);
+
+    const response = await zoomRetry(
+        () =>
+            fetch(`${zoomMeetingUrl}/${meetingId}`, {
+                method: 'PATCH',
+                headers: {
+                    Authorization: `Bearer ${access_token}`,
+                    'Content-Type': 'application/json',
+                },
+                body: body,
+            }),
+        3,
+        1000
+    );
+
+    if (!response.ok && response.status !== 404) {
+        throw new Error(`Zoom - failed to update meeting with ${response.status} ${await response.text()}`);
+    }
+
+    if (response.status === 204) {
```

Comment:
Why do we conditionally log? What if the response is ok but not 204? Why wouldn't we log in that case?